src/Meshes/interval.jl

```diff
@@ -167,34 +191,48 @@ function IntervalMesh(
     domain::IntervalDomain{CT},
     stretch::GeneralizedExponentialStretching{FT};
     nelems::Int,
+    reverse_mode::Bool = false,
 ) where {CT <: Geometry.Abstract1DPoint{FT}} where {FT}
     if nelems ≤ 1
         throw(ArgumentError("`nelems` must be ≥ 2"))
     end
+
     dz_bottom, dz_top = stretch.dz_bottom, stretch.dz_top
-    if !(dz_bottom ≤ dz_top)
+    if !(dz_bottom ≤ dz_top) && !reverse_mode
         throw(ArgumentError("dz_bottom must be ≤ dz_top"))
     end
-    # surface coord height value
+
+    if !(dz_bottom ≥ dz_top) && reverse_mode
+        throw(ArgumentError("dz_top must be ≤ dz_bottom"))
+    end
+
+    # bottom coord height value, always min, for both atmos and land, since z-axis does not change
     z_bottom = Geometry.component(domain.coord_min, 1)
-    # top coord height value
+    # top coord height value, always max, for both atmos and land, since z-axis does not change
     z_top = Geometry.component(domain.coord_max, 1)
+    # but in case of reverse_mode, we temporarily swap them together with dz_bottom and dz_top
+    # so that the following root solve algorithm does not need to change
+    if reverse_mode
+        z_bottom, z_top = Geometry.component(domain.coord_max, 1),
+        -Geometry.component(domain.coord_min, 1)
+        dz_top, dz_bottom = dz_bottom, dz_top
+    end
 
     # define the inverse σ⁻¹ exponential stretching function
-    exp_stretch(ζ, h) = -h * log(1 - (1 - exp(-1 / h)) * ζ)
+    exp_stretch(ζ, h) = ζ == 1 ? ζ : -h * log(1 - (1 - exp(-1 / h)) * ζ)
```

Comment:
What is this change needed for? Is this to mirror what's done in the exponential stretching?

Reply:
Yes, because otherwise the corner case of the extremum might have not be exactly computed